### PR TITLE
Humble / Python 3.10

### PR DESCRIPTION
**Pull request by InvincibleRMC**
# Pull Request

## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Bug Fix
- [x] Optimization
- [x] Documentation Update

## Description

Dear Benjamin,

I hope this markdown finds in you good health. Moving to Humble and Python 3.10 has brought several nice new features especially in the typing department. And has only broken one thing. The submodule we use for i2c communication has an import which no longer exists. However, we were planning on deprecating it anyway as of #5 so as of now it remains unfixed for rookies to work on. This branch is also the foundation of the mavros switch but, figured it would be better left in it's own pr #15. Things that have not worked great in the past have not been super thoroughly tested ej. cameras. I have gone through and used isort so our imports are more professional. I have added a building hotkey for our easy build. Also fixed a janky bug with /task_request service.

TLDR:
FIxed stuff I'm sure we will have a long discussion about all 50 changed files

Best,
Michael Carlstrom

## Related Tickets & Documents

- Related Issue #
#5, #15 
- Closes #
#1

### CODE REVIEW

File: .vscode/easy_all.sh [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Done to suppress setup.py warnings
```
Comment (benjaminwp18) [inline .vscode/easy_all.sh:1]: Mention the fact that ROS hasn't updated their packages to use the new build system yet

File: .vscode/easy_build.sh [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Done to suppress setup.py warnings
```
Comment (benjaminwp18) [inline .vscode/easy_build.sh:1]: Mention the fact that ROS hasn't updated their packages to use the new build system yet

File: src/surface/surface_main/scripts/ROS2-Install-Humble.sh [hunk 3]
```diff
@@ -36,14 +36,33 @@ echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-a
...
-fi
\ No newline at end of file
+ROS_LINE='source /opt/ros/humble/setup.bash'
+if ! grep -qF "$ROS_LINE" ~/.bashrc ; 
+    then echo "$ROS_LINE" >> ~/.bashrc ;
+fi
+
+# Done to suppress setup.py install depracted warnings
+# Hopefully in the next ROS release it will not be as janky
+# Seems to work fine however
```
Comment (benjaminwp18) [inline src/surface/surface_main/scripts/ROS2-Install-Humble.sh:58]: Mention the fact that ROS hasn't updated their packages to use the new build system yet
Also fix "depracted" typo

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/keyboard_driver/keyboard_driver/keyboard_driver_node.py [hunk 4]
```diff
@@ -84,37 +88,45 @@ def logger(self):
...
+                case Keys.PITCH_DOWN:
+                    self.status["pitch_down"] = True
+                case Keys.YAW_LEFT:
+                    self.status["yaw_left"] = True
+                case Keys.YAW_RIGHT:
+                    self.status["yaw_right"] = True
+                case Keys.HELP:
+                    self.logger.info(Keys.HELP_MSG)
+                case _:
+                    return
```
Comment (benjaminwp18) [inline src/surface/keyboard_driver/keyboard_driver/keyboard_driver_node.py:129]: If we're already refactoring the keys, this conversion should probably happen in Keys. Either a method that accepts a key and spits out a status, or a dictionary that does the same thing.

File: src/pi/pixhawk_communication/pixhawk_communication/pixhawk_communication.py [hunk 5]
```diff
@@ -25,13 +26,13 @@ def __init__(self):
...
             self.arm_callback,
-            1
+            qos_profile_system_default
         )
         self.rov_control_sub: Subscription = self.create_subscription(
             ROVControl,
             'manual_control',
             self.rov_control_callback,
-            100
+            qos_profile_system_default
```
Comment (benjaminwp18) [inline src/pi/pixhawk_communication/pixhawk_communication/pixhawk_communication.py:35]: Is it fine for these to be the same? I can't test this bit

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/pi/pixhawk_communication/pixhawk_communication/pixhawk_communication.py (hunk 5 above)
Comment (InvincibleRMC) [reply src/pi/pixhawk_communication/pixhawk_communication/pixhawk_communication.py:35]: huh?

**Review by benjaminwp18 — APPROVED** [review]
